core/src/main/java/org/infinispan/protostream/annotations/impl/AbstractMarshallerCodeGenerator.java

```diff
@@ -719,7 +721,11 @@ protected void generateWriteMethodBody(IndentWriter iw, ProtoMessageTypeMetadata
                         mapFieldMetadata.getValue().getJavaTypeName()
                   );
                } else {
-                  iw.printf("java.util.Collection<%s>", fieldMetadata.getJavaTypeName());
+                  if (useGenerics) {
```

Comment:
This is always false.

Reply:
I know, I wanted to keep the code around for when I do it properly